src/machine.rs

```diff
@@ -272,8 +272,8 @@ impl<'mir, 'tcx> Machine<'mir, 'tcx> for Evaluator<'tcx> {
     }
 
     #[inline(always)]
-    fn abort(_ecx: &mut InterpCx<'mir, 'tcx, Self>) -> InterpResult<'tcx, !> {
-        throw_machine_stop!(TerminationInfo::Abort)
+    fn abort(_ecx: &mut InterpCx<'mir, 'tcx, Self>, msg: Option<String>) -> InterpResult<'tcx, !> {
+        throw_machine_stop!(TerminationInfo::Abort(msg.unwrap_or_default()))
```

Comment:
This will look really odd when there is no message:
```
the evaluated program aborted execution: 
```
I suggest to make `TerminationInfo::Abort` take `Option<String>`, and then just throw with `None` here.

Reply:
hmm, and then print `Some("attempted to...")` when there's an error?

also, if I replace to `throw_machine_stop!` I'll avoid touching this function.
although I think it will also be nice for const-eval in the future

Reply:
With a message, just add the `: ` and then the error message.

> although I think it will also be nice for const-eval in the future

Then we can change it in the future, if it really turns out to be useful. Let's avoid adding API surface now that will bitrot and remain unused.